### PR TITLE
Integrate with Circle CI

### PR DESCRIPTION
**Pull request by Pradeepkumarbk**
- Currently, when a commit happens, travis  gets triggered.
But, due to frequent issues with travis and other debugging issues with it, circleCI integration is needed.
- In Circle CI we have a time limit of 120mins for each build.
- Admin can ssh into the build and debug it and also after the build is completed the VM will be active for 30minutes for debugging .
- Due to kernel version issue only build will happen for build-tag-0.
- Deactivated Travis ci when PR is raised, when PR is raised CircleCI is triggered and when merged Travis is triggered. 

Signed-off-by: Pradeepkumarbk <pradeepkumar.bk96@gmail.com>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            sudo cp *.a /usr/lib
+            popd
+            cd ..
+
+            # we need fio repo to build zfs replica fio engine
+            git clone https://github.com/axboe/fio
+            cd fio
+
+            # TODO replace commit hash by version tag in future
+            git checkout 2e4ef4fbd69eb6d4c07f2f362463e3f3df2e808c
```
Comment (jkryl) [inline .circleci/config.yml]: there was a recent change to .travis.yml which should be done here as well:
```
-    # TODO: replace commit hash by version tag in future
-    - git checkout 2e4ef4fbd69eb6d4c07f2f362463e3f3df2e808c
+    - git checkout fio-3.7
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            git checkout spl-0.7.4
+            sh autogen.sh
+            ./configure
+            if [ "$ZFS_BUILD_TAGS" = "0" ]; then
+              make -j4;
+            fi
+            cd ../zfs
+            sh autogen.sh
+            if [ "$ZFS_BUILD_TAGS" = "0" ]; then
+              ./configure --with-config=user --enable-code-coverage=yes --enable-debug --enable-uzfs=yes --with-jemalloc --with-fio=$PWD/../fio || true;
```
Comment (jkryl) [inline .circleci/config.yml]: why `|| true` at the end?

**Review by jkryl — COMMENTED** [review]
Thanks for the PR! I'm curious what are we going to do with kernel build? Without it ZFS_BUILD_TAGS are useless and can be removed.

File: .circleci/config.yml (hunk 1 above)
Comment (Pradeepkumarbk) [reply .circleci/config.yml]: Addressed

**Review by jkryl — APPROVED** [review]

**Comment by vishnuitta** [comment]
@Pradeepkumarbk currently, this PR is merged.
We need to handle failure cases i.e., if build fails or testcase fails, build need to be marked as failure.
Are we handling failure cases now?

**Pull request merged**